web/modules/custom/usagov_ssg_postprocessing/src/EventSubscriber/PublishedPagesSubscriber.php

```diff
@@ -32,7 +32,22 @@ public function modifyHtml(ModifyHtmlEvent $event) {
     $csv = [];
     $fp = fopen($csv_path, 'c+');
     if ($fp != FALSE) {
-      flock($fp, LOCK_EX);
+
+      $locks = 0;
+      // Wait of 0.01 seconds to get a lock. Try up to 50 times.
+      // Failure to get a lock might is likely extremely rare, but more likely
+      // if using more tome processes to export a file. If we don't get a
+      // lock here, we may overwrite another processes' changes to the CSV.
+      while (FALSE === flock($fp, LOCK_EX)) {
```

Comment:
I have a question! When @jacobaaronyeager and I put this locking in, we struggled with whether we needed to check and retry like this, and we concluded that we didn't. https://www.php.net/manual/en/function.flock.php says "By default, this function will block until the requested lock is acquired [...]" 

And we did some light experimentation and concluded that was true -- if a file was locked, flock() didn't return until the existing lock was released. 

Are you actually finding that flock() returns false rather than blocking? I can certainly believe we missed something.

Reply:
I was confused by flocks versus other mutexes and the old comments on that page. Also from other reading, flock is only an exclusive lock within a process. There may still be a potential issue, but in real-world runs, this may happen so quickly that it's not an issue.

I've reverted this code change.

Reply:
The other thing that's annoying me is that tome says its exporting 1820ish pages, but the CSV file has about 200 fewer entries than that. It looks like that initial total includes taxonomy and user paths that aren't actually exported.

Reply:
I thought flock was exclusive across processes too, at least on linux. I should try harder to get something definitive on that. 

Regarding the page count, I'm sure it includes pages that are later excluded by some custom code I put in to exclude whole directories ... which probably includes close to 200 pages of jsonapi paths.